server/data/hmppsAuthClient.ts

```diff
@@ -71,14 +83,23 @@ export default class HmppsAuthClient {
     return this.restClient(token).get({ path: '/api/user/me' }) as Promise<User>
   }
 
-  getUserByEmailAddress(token: string, emailAddress: string): Promise<User> {
-    logger.info(`Getting user detail by email address: calling HMPPS Auth`)
-    return this.restClient(token).get({ path: `/api/authuser`, query: { email: emailAddress } }) as Promise<User>
+  async getUserByEmailAddress(token: string, emailAddress: string): Promise<AuthUser> {
+    logger.info(`Getting auth user detail by email address: calling HMPPS Auth`)
+    const authUsers = (await this.restClient(token).get({
+      path: `/api/authuser`,
+      query: { email: emailAddress },
+    })) as AuthUser[]
+
+    return authUsers[0]
   }
 
-  getUserByUsername(token: string, username: string): Promise<User> {
+  async getUserByUsername(token: string, username: string): Promise<AuthUser> {
     logger.info(`Getting user detail by username: calling HMPPS Auth`)
-    return this.restClient(token).get({ path: `/api/authuser/${username}` }) as Promise<User>
+    const authUsers = (await this.restClient(token).get({
+      path: `/api/authuser/${username}`,
+    })) as AuthUser[]
```

Comment:
this endpoint returns a single value, not a list.

Reply:
Oops, well spotted. Have updated this